transcom/mymove
# [ MB-16671 ] Use sit origin hhg original address for origin sit service area payment

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16671)

## Summary

This PR updates the calculations for the Service Area Origin param so that it is utilizing the SIT origin HHG original address instead of the shipment's pickup address for origin SIT service items. 

I modeled the change off of [this PR](https://github.com/transcom/mymove/pull/10918), which does something similar for destination SIT service items. 

I also updated the ZipPickup param to reflect the SIT origin HHG original address for origin SIT service items as well.  

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test
Okay so this is a little complicated. We are basically testing that we are using the service_area associated with the `SITOriginHHGOriginalAddress` and not the service area associated with the `SITOriginHHGActualAddress`. When you create a SIT service item, the value supplied in the actual address is copied to the shipment pickup address as well. 

So you want to make a move that has an original pickup address that has a different service area from the `SITOriginHHGActualAddress` you provide when creating the origin SIT service item.

I used the following:
pickup zip 90210 is in service area `056`
`SITOriginHHGActualAddress` zip 12345 is in `536`

If you would like you can use the following query to verify other zips on your own:
```SQL
select service_area from re_domestic_service_areas rdsa
join re_zip3s rz3 on rz3.domestic_service_area_id = rdsa.id
	where rz3.zip3 = '<first 3 digits of the zipcode>';
```

1. Create a move as a customer or find a new move as a TOO. Most generated moves will be from pickup zip 90210. 
2. Login as a TOO and make the move available to the prime. You may need to add info to the orders for it to be approvable.
3. Log in to the prime simulator and create the origin SIT service items. Make sure to provide a `SITOriginHHGActualAddress` that has a zip code that is in a different service area than the initial pickup address.
4. Log back in as a TOO and approve the service items.
5. Log back in as the prime and create a payment request for the origin SIT service items
6. Log back in as a TIO and review the service items. 
7. View the Domestic origin add't SIT and the first day SIT service items. Click on 'show calculations' and you should see the SIT Origin service area display the correct service area
![image](https://github.com/transcom/mymove/assets/110134470/77fbe6f1-aa86-4c19-a445-9b9e3571697f)


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
